docs/pages/faq/deferred-computation.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Deferred Computation in Expo and React Native
+---
+
+import SnackInline from '~/components/plugins/SnackInline'
+
+When building an application, you will often need to perform some computation after a period of time rather than in response to some event or component lifecycle event.  Because mobile applications rely heavily on interactions and your computation shares a JS thread with updates to the UI, performant apps must be careful to schedule work in small increments and at times that will not impact your users' experience.
+
+
+## While your app is in the foreground
+ While your application is in the foreground, you have access to all of the techniques available in JS development like setTimeout, setInterval, and requestAnimationFrame.  If you are not familiar with these methods or when you might use them we recommend Mozilla's guide on [asynchronous Javascript](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Timeouts_and_intervals).
+
+React Native provides an additional timing technique with InteractionManager.  InteractionManager allows you to schedule computationally expensive computations to run after any interactions and animations they might impact have completed.  You can schedule computations to run with `InteractionManager.runAfterInteractions(() => { /* your task */})` React Native itself, and high-quality libraries you might use, already register themselves with `InteractionManager`.  If you need to, you can register any other animations via `InteractionManager.createInteractionHandle()` on the start of your animation and signal completion via `InteractionManager.clearInteractionHandle(resultOfCallToCreateInteractionHandle)`
+
+
+### Usage in React Native
+
+In React Native You may need to update your application's state as a result of your computation.  A few points to note:
+- Make sure that your timers do not live beyond the lifecycle of your component by clearing your timers created in componentWillUnmount.
+- You must bind the context, either explicitly or by using arrow notation, to have access to methods like setState when your computation is run.
+- setState is asynchronous, so if you need to mutate the existing state, pass a function rather than an object to the method.
```

Comment:
```suggestion
- `setState` is asynchronous, so if you need to mutate the existing state, pass a function rather than an object to the method.
```
Is there a page in the React docs we could link to?